File: leetcode_solutions/string/areOccurenceEqual.java
public class areOccurrencesEqual {
    public boolean areOccurrencesEqual(String s) {
        HashMap<Character, Integer> frequencyMap = new HashMap<>();
        for(char c : s.toCharArray()) {
            // buyerda update qilamiz
            frequencyMap.put(c, frequencyMap.getOrDefault(c, 0) + 1);
        }
        // birinchi frequencyni olamiz
        int firstFrequency = frequencyMap.values().iterator().next();
        for(int frequency: frequencyMap.values()) {
            if(frequency != firstFrequency){
                // agar bironta freq boshqacha bo'lsa false qaytaramiz
                return false;
            }
        }
        return true;
    }
}

File: leetcode_solutions/string/areOccurenceEqual.ts
function areOccurrencesEqual(s: string): boolean {
    const frequencyMap = new Map<string, number>();
    for(const character of s) {
        // update qilamiz frequency countni
        frequencyMap.set(character, (frequencyMap.get(character) || 0) + 1);
    }
    // first frequency value
    const firstFrequency = frequencyMap.values().next().value;
    // har bir frequencyni tekshirib chiqamiz
    for(const frequency of frequencyMap.values()){
        if(frequency != firstFrequency){
            return false;
        }
    }
    return true;
};